test/Constraints/bidirectional_conversions.swift

```diff
@@ -0,0 +1,113 @@
+// RUN: %target-typecheck-verify-swift
+// REQUIRES: objc_interop
+
+import Foundation
+import CoreGraphics
+
+/////////////
+
+struct G<T> {  // expected-note {{arguments to generic parameter 'T' ('CGFloat?' and 'CGFloat') are expected to be equal}}
+  var t: T
+}
+
+func foo1(x: (x: Int, y: Int)?, y: (Int, Int)) -> G<(x: Int, y: Int)> {
+  let g = G(t: x ?? y)
```

Comment:
This was the specific case that regressed with the deep equality change, but if you look at the rest of the file you'll see the general behavior with this sort of thing is quite inconsistent...